src/handlers/resources/ResourcesStreamHandler.cpp

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "ResourcesStreamHandler.h"
+#include "ThreadMonitor.h"
+
+namespace visor::handler::resources {
+
+ResourcesStreamHandler::ResourcesStreamHandler(const std::string &name, InputStream *stream, const Configurable *window_config, StreamHandler *handler)
+    : visor::StreamMetricsHandler<ResourcesMetricsManager>(name, window_config)
+    , _timer(time(NULL))
+{
+    if (handler) {
+        throw StreamHandlerException(fmt::format("ResourcesStreamHandler: unsupported upstream chained stream handler {}", handler->name()));
+    }
+
+    assert(stream);
+    // figure out which input stream we have
+    if (stream) {
+        _pcap_stream = dynamic_cast<PcapInputStream *>(stream);
+        _dnstap_stream = dynamic_cast<DnstapInputStream *>(stream);
+        _mock_stream = dynamic_cast<MockInputStream *>(stream);
+        _sflow_stream = dynamic_cast<SflowInputStream *>(stream);
+        if (!_pcap_stream && !_mock_stream && !_dnstap_stream && !_sflow_stream) {
+            throw StreamHandlerException(fmt::format("NetStreamHandler: unsupported input stream {}", stream->name()));
+        }
+    }
+}
+
+void ResourcesStreamHandler::start()
+{
+    if (_running) {
+        return;
+    }
+
+    if (config_exists("recorded_stream")) {
+        _metrics->set_recorded_stream();
+    }
+
+    if (_pcap_stream) {
+        _pkt_connection = _pcap_stream->packet_signal.connect(&ResourcesStreamHandler::process_packet_cb, this);
+    } else if (_dnstap_stream) {
+        _dnstap_connection = _dnstap_stream->dnstap_signal.connect(&ResourcesStreamHandler::process_dnstap_cb, this);
+    } else if (_sflow_stream) {
+        _sflow_connection = _sflow_stream->sflow_signal.connect(&ResourcesStreamHandler::process_sflow_cb, this);
+    }
+
+    _running = true;
+}
+
+void ResourcesStreamHandler::stop()
+{
+    if (!_running) {
+        return;
+    }
+
+    if (_pcap_stream) {
+        _pkt_connection.disconnect();
+    } else if (_dnstap_stream) {
+        _dnstap_connection.disconnect();
+    } else if (_sflow_stream) {
+        _sflow_connection.disconnect();
+    }
+
+    _running = false;
+}
+
+void ResourcesStreamHandler::process_sflow_cb([[maybe_unused]] const SFSample &)
+{
+    if (difftime(time(NULL), _timer) >= MEASURE_INTERVAL) {
```

Comment:
used time_t to calculate diff time to get resources in the fixed interval (10 seconds)
the idea to use time_t is that we don't need precision here and it is faster then clocks methods:
https://gist.github.com/stunpix/54c304bc010307562ec3434dbd5e76ae